### PR TITLE
Fix OP_NEG operation and flags in ALU (provided by yume)

### PR DESCRIPTION
**Pull request by Plaque-fcc**
Signed-off-by: Plaque-fcc <Plaque-fcc@github>

### CODE REVIEW

File: risc8-alu.v [hunk 1]
```diff
@@ -191,8 +191,8 @@ module risc8_alu(
 `endif
 `ifdef CONFIG_OP_NEG
 		`OP_NEG: begin
-			R = ~Rd;
-			SH = R3 | !Rd3;
+			R = -Rd;
```
Comment (osresearch) [inline risc8-alu.v:194]: Oops.  Yes, `-Rd` or `8'h00 - Rd` should be here.

File: risc8-alu.v [hunk 2]
```diff
@@ -191,8 +191,8 @@ module risc8_alu(
...
+			SH = (R3 & !Rd3) | Rd3;
```
Comment (osresearch) [inline risc8-alu.v]: image

The instruction set manual says `H = P3 + !Rd3`; I had assumed that `P3` was a typo for `R3`.

`SH = (R3 & !Rd3) | Rd3` distributes to `(R3 | Rd3) & (!Rd3 | Rd3) == R3 | Rd3`, which doesn't seem right.

Comment (Plaque-fcc) [reply risc8-alu.v]: Feels like the document isn't right about how it works. ;D
We're thinking of a replacement.

Comment (Plaque-fcc) [reply risc8-alu.v]: Apparently, in [AVR Instruction Set Manual](https://ww1.microchip.com/downloads/en/DeviceDoc/AVR-Instruction-Set-Manual-DS40002198A.pdf) © 2020 Microchip Technology Inc., 5.80.2, they say exactly this:

### H R3∨Rd3
> Set if there was a borrow from bit 3; cleared otherwise.

Which means `SH = R3 | Rd3` is actually correct.

**Comment by Plaque-fcc** [comment]
Agreed. I've replaced the commit.

**Comment by Plaque-fcc** [comment]
From the author of all these patches:

The document "atmel-0856-avr-instruction-set-manual.pdf" contains a lot of errors, for example NEG and SBIW, I'm sure there's more of them.
Please don't use it as a reference, try [AVR-Instruction-Set-Manual-DS40002198A.pdf](https://ww1.microchip.com/downloads/en/DeviceDoc/AVR-Instruction-Set-Manual-DS40002198A.pdf) instead.
My code was initially wrong in that line, it should be "R3 | Rd3", but "(R3 & !Rd3) | Rd3" also works since it's the same.
I've tried various variants until the output started matching the results I was getting from simulavr, and for some reason it ended as "(R3 & !Rd3) | Rd3".
Of course this can easily be replaced with "R3 | Rd3" for simplicity and clarity.

_Unedited part:_
```
Now explanation:
Operation NEG is: Rd = 0 - Rd
Now look at OP_SUB and consider (Rd = 0, Rr = some value):
SH = (!Rd3 & Rr3) | (Rr3 & R3) | (R3 & !Rd3);
Since Rd is 0, then Rd3 is also 0, and !Rd3 is 1
SH = (1 & Rr3) | (Rr3 & R3) | (R3 & 1);
SH = Rr3 | (Rr3 & R3) | R3;
Since (Rr3 & R3) is pointless we remove it and get:
SH = Rr3 | R3;
Since for NEG source is Rd, then Rr3 is replaced to Rd3
```

In short, in the latest specs it is described correctly. It's gonna take some time to verify it on a real hardware if you doubt that ```SH = R3 | Rd3``` is right. He thinks that the explanation above should be enough, tho.

**Comment by osresearch** [comment]
Thanks for the updated reference -- I had been using 0856 (from 2016), and had almost been alerted to the `SBIW` error in the past.  The 2020 version looks better and fixes both the typo for `R3` as well as the logic.
